pkg/helmreconciler/rendering.go

```diff
@@ -38,6 +37,14 @@ import (
 	"istio.io/operator/pkg/validate"
 	binversion "istio.io/operator/version"
 	"istio.io/pkg/log"
+	"istio.io/pkg/version"
+)
+
+var (
+	// objectCache holds the latest copy of each object applied by the controller, keyed by its Hash() function.
+	// Must be locked since ProcessManifest can be run concurrently.
+	objectCache   = make(map[string]*object.K8sObject)
+	objectCacheMu sync.RWMutex
```

Comment:
How does this global cache work with multi-tenant controllers? I think it needs to be key'd by ICP (namespace). Alternately, the cache is attached to the reconciler itself which seems to be per-ICP instance already. The latter option would be preferred.

Reply:
Yes, it should be keyed by ICP resource.  For multi-tenancy, it's possible a single controller will be managing all installs (i.e. watching all namespaces).

Reply:
Even with multi-tenant, would we expect to see overlapping objects coming from different ICPs? That seems like it shouldn't be allowed. Even with no caching it would lead to non-deterministic behavior.

Reply:
Oh, I see namespace is part of the key and presumably there is only one ICP install per namespace.

Reply:
Right.